Octokit.Reactive/Clients/IObservableTeamDiscussionsClient.cs

```diff
@@ -0,0 +1,88 @@
+        /// <param name="teamId">The team identifier.</param>
+        /// <param name="number">The team discussion number.</param>
+        /// <returns>The <see cref="TeamDiscussion"/> with the given identifier.</returns>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        IObservable<TeamDiscussion> Get(int teamId, int number);
+
+        /// <summary>
+        /// List all discussions on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/teams/discussions/#list-discussions
+        /// </remarks>
+        /// <param name="teamId">The team identifier.</param>
+        /// <returns>A list of discussions in the team.</returns>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        IObservable<TeamDiscussion> GetAll(int teamId);
+
+        /// <summary>
+        /// List all discussions on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// https://developer.github.com/v3/teams/discussions/#list-discussions
+        /// </remarks>
+        /// <param name="teamId">The team identifier.</param>
+        /// <param name="options">Options to change API behaviour.</param>
+        /// <returns>A list of discussions in the team.</returns>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        IObservable<TeamDiscussion> GetAll(int teamId, ApiOptions options);
+
+        /// <summary>
+        /// Creates a new discussion post on a team's page.
+        /// </summary>
+        /// <remarks>
+        /// OAuth access tokens require the write:discussion scope.
+        /// https://developer.github.com/v3/teams/discussions/#create-a-discussion
+        /// </remarks>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        /// <returns>Newly created <see cref="TeamDiscussion"/> object.</returns>
+        IObservable<TeamDiscussion> Create(int teamId, NewTeamDiscussion teamDiscussion);
+
+        /// <summary>
+        /// Edits the title and body text of a discussion post.
+        /// Only the parameters you provide are updated.
+        /// </summary>
+        /// <remarks>
+        /// OAuth access tokens require the write:discussion scope.
+        /// https://developer.github.com/v3/teams/discussions/#edit-a-discussion
+        /// </remarks>
+        /// <param name="teamId">The team identifier.</param>
+        /// <param name="number">The discussion number which will be update with new values.</param>
+        /// <param name="teamDiscussion">New values for the discussion.</param>
+        /// <exception cref="ApiException">Thrown when a general API error occurs.</exception>
+        /// <returns>Updated <see cref="TeamDiscussion"/> object.</returns>
+        IObservable<TeamDiscussion> Update(int teamId, int number, UpdateTeamDiscussion teamDiscussion);
```

Comment:
can we please name the variable `UpdateTeamDiscussion updateTeamDiscussion` here and in the implementation?

```suggestion
        IObservable<TeamDiscussion> Update(int teamId, int number, UpdateTeamDiscussion updateTeamDiscussion);
```